File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = [0] * 26
        l, max_length, max_freq = 0, 0, 0

        for r in range(len(s)):
            count[ord(s[r]) - ord('A')] += 1
            max_freq = max(max_freq, count[ord(s[r]) - ord('A')])

            while r - l + 1 - max_freq > k:
                count[ord(s[l]) - ord('A')] -= 1
                l += 1
            
            max_length = max(max_length, r - l + 1)
        
        return max_length

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0, max_count = 0, max_freq = 0;
        std::array<int, 26> count = {0};
        
        for (int r = 0; r < s.length(); r++) {
            count[s[r] - 'A']++;
            max_freq = std::max(max_freq, count[s[r] - 'A']);
            while (r - l + 1 - max_freq > k) {
                count[s[l] - 'A']--;
                l++;
            }
            max_count = std::max(max_count, r - l + 1);
        }

        return max_count;
    }
};